Seqexec-Web first batch

cquiroz opened the pull request:
This PR brings the "experimental" seqexec-web work done so far. It builds a mostly static web ui using React.js and scala.js. For the UI [SemanticUI](http://semantic-ui.com/) has been selected even though there are no react bindings. They will be done in this project while we progress on the UI

There is a ton of work pending but this establishes a start point to publish the work done so far. At this point the UI is not doing much but displays a queue based on canned data 

![screenshot 2016-03-29 12 48 45](https://cloud.githubusercontent.com/assets/3615303/14114247/a9e6b198-f5ac-11e5-8fbb-de284e2730b6.png)

swalker2m commented on modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/semanticui/elements/icon/Icon.scala:
```diff
@@ -0,0 +1,73 @@
+                   circular: Boolean = false,
+                   bordered: Boolean = false,
+                   inverted: Boolean = false,
+                   color: Option[String] = None)
```
I take it these are attributes on some given UI component that we're giving a type?  Since the model supports it, what would it mean to be tiny, but not mini, but small, but not large, but huge but not massive?  Should this shunted into case objects like you did with `Flipped` and `Rotated`?

cquiroz replied:
I guess for the size we could make it into objects. In general they are booleans that decide the presence of a given css class. In principle you could make an Icon both `tiny massive` though it probably make sense to forbid it. All the other attributes seem to be independent

swalker2m commented on modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/mdl/MaterialComponent.scala:
```diff
@@ -0,0 +1,38 @@
+        js.Dynamic.global.componentHandler.upgradeElement(children(i))
+      )
+    }
+  }
```
Would you add a comment or two here maybe?  I'm not sue what is going on.

cquiroz replied:
This code is unused as I'm not using MDL anymore but I wanted to keep it just in case
It essentially calls the MDL function `componentHandler.upgradeElement` on the current DOM node and its children 
I'll add some comments though

cquiroz commented:
Updated according to comments on the PR

cquiroz commented:
Please ignore the travis failure, it is not setup properly yet

arturog8m commented on modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/README.md:
```diff
@@ -0,0 +1,64 @@
+
+## TODO
+
+* Authentication
```
Do you really want to keep and maintain a todo list here?

cquiroz replied:
Right, those are from the original experiment, I'll convert them to GH issues

arturog8m commented on modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/resources/application.conf:
```diff
@@ -0,0 +1,252 @@
+#
+# Copyright (C) 2009-2016 Typesafe Inc. <http://www.typesafe.com>
+#
```
What are the implications of this copyright?

cquiroz replied:
I don't think it  is too relevant, this is just a configuration file. Actually I hoped that it could be read directly from the play jars, Maybe I'll try that and try removing this conf

cquiroz commented:
Addressed both @arturog8m questions, Removed the TODOs on the `README.md` files and converted them into GH issues and reduced the size of the play configuration removing the default bits

tpolecat commented on modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/README.md:
```diff
@@ -0,0 +1,28 @@
+
+* [react.js](https://facebook.github.io/react/) Already described
+* [Diode](https://github.com/ochrons/diode) For data flow and organization of the application
+* [uPickle](https://github.com/japgolly/upickle): Used for json serialization. Even though we have Argonaut already, uPickle runs on both sides of the equation. We may decide to use [BooPickle](https://github.com/ochrons/boopickle) instead for better performance
```
Why are we using japgolly's version rather than lihaoyi's?

tpolecat replied:
Oh that looks like a typo.

cquiroz replied:
Right, it is a typo

tpolecat commented:
:+1:
